src/Resources/skeleton/forgottenPassword/ForgottenPasswordController.tpl.php

```diff
@@ -0,0 +1,110 @@
+            if (!$user) {
+                return $this->redirectToRoute('app_check_email');
+            }
+
+            // generate a reset password token, that the user could use to change his password
+            $resetPasswordToken = new <?= $token_class_name?>($user);
+            $this->getDoctrine()->getManager()->persist($resetPasswordToken);
+            $this->getDoctrine()->getManager()->flush();
+
+            $message = (new \Swift_Message('Your password reset request'))
+                ->setFrom(['noreply@mydomain.com' => 'Noreply'])
+                ->setTo($user-><?= $email_getter ?>())
+                ->setBody($this->renderView('forgottenPassword/email.txt.twig', [
+                    'token' => $resetPasswordToken,
+                ]))
+            ;
+            $mailer->send($message);
+
+            return $this->redirectToRoute('app_check_email');
+        }
+
+        return $this->render('forgottenPassword/request.html.twig', [
+            'requestForm' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/check-email", name="app_check_email")
+     */
+    public function checkEmail()
+    {
+        return $this->render('forgottenPassword/check_email.html.twig', [
+            'tokenLifetime' => <?= $token_class_name ?>::TTL_HOURS,
+        ]);
+    }
+
+    /**
+     * @Route("/reset-password/{tokenAndSelector}", name="app_reset_password")
+     */
+    public function reset(Request $request, $tokenAndSelector, UserPasswordEncoderInterface $passwordEncoder): Response
+    {
+        $passwordResetToken = $this->getDoctrine()->getRepository(PasswordResetToken::class)->findOneBy([
+            'selector' => substr($tokenAndSelector, 0, PasswordResetToken::SELECTOR_LENGTH),
+        ]);
+
+        if (!$passwordResetToken) {
+            throw $this->createNotFoundException();
+        }
+        if ($passwordResetToken->isExpired() || !$passwordResetToken->isTokenEquals(substr($tokenAndSelector, PasswordResetToken::SELECTOR_LENGTH))) {
+            $this->getDoctrine()->getManager()->remove($passwordResetToken);
+            $this->getDoctrine()->getManager()->flush();
+
+            throw $this->createNotFoundException();
+        }
+
+        $form = $this->createForm(<?= $resetting_form_class_name ?>::class);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            // a PasswordResetToken should be used only once, remove it
+            $this->getDoctrine()->getManager()->remove($passwordResetToken);
+            $this->getDoctrine()->getManager()->flush();
+
+            // encode the plain password, and set it
+            $passwordResetToken->getUser()-><?= $password_setter ?>(
+                $passwordEncoder->encodePassword(
+                    $passwordResetToken->getUser(),
+                    $form->get('plainPassword')->getData()
+                )
+            );
```

Comment:
And then we need one more `flush()` then for the new password... or move token's `remove()` and `flush()` after `setPasswrod()` call?

Reply:
Correct, my modification broke it! So actually I would rather keep the `flush()` at the previous position.

Reply:
(so everything is done in one "DB" operation)

Reply:
Modification done!